programs/mango-v4/src/instructions/token_conditional_swap_start.rs

```diff
@@ -0,0 +1,141 @@
+use anchor_lang::prelude::*;
+use fixed::types::I80F48;
+
+use crate::accounts_ix::*;
+use crate::error::*;
+use crate::health::*;
+use crate::i80f48::ClampToInt;
+use crate::logs::TokenBalanceLog;
+use crate::logs::TokenConditionalSwapStartLog;
+use crate::state::*;
+
+#[allow(clippy::too_many_arguments)]
+pub fn token_conditional_swap_start(
+    ctx: Context<TokenConditionalSwapStart>,
+    token_conditional_swap_index: usize,
+    token_conditional_swap_id: u64,
+) -> Result<()> {
+    let group_pk = &ctx.accounts.group.key();
+    let account_key = ctx.accounts.account.key();
+    let caller_key = ctx.accounts.caller.key();
+    require_keys_neq!(account_key, caller_key);
+
+    let mut account = ctx.accounts.account.load_full_mut()?;
+    require!(
+        !account.fixed.being_liquidated(),
+        MangoError::BeingLiquidated,
+    );
+
+    let mut caller = ctx.accounts.caller.load_full_mut()?;
+
+    let mut account_retriever = ScanningAccountRetriever::new(ctx.remaining_accounts, group_pk)
+        .context("create account retriever")?;
+
+    let tcs = account
+        .token_conditional_swap_by_index(token_conditional_swap_index)?
+        .clone();
+    require!(tcs.has_data(), MangoError::TokenConditionalSwapNotSet);
```

Comment:
can we rename the has_data to is_active? that would be consistent with other things we have no? well the only difference is that it can be expired I guess, but has_data seems to low level imo

Reply:
I agree about this being inconsistent with other components, but `is_active` seemd confusing to me: It could be expired or it could be an auction that hasn't started yet.

`is_set`, `is_configured`, `is_defined`?

Reply:
is_configured lgtm